src/web/browser/coreVitals/coreVitals.ts

```diff
@@ -0,0 +1,74 @@
+import { getCLS, getFID, getLCP, getFCP, getTTFB } from 'web-vitals';
+
+// Stops ts assuming json values are null
+const temp: any = null;
+
+const jsonData = {
+	page_view_id: '',
+	received_timestamp: `${new Date().toISOString()}`,
+	id: 'prod',
+	received_date: new Date().toISOString().slice(0, 10),
+	fid: temp,
+	cls: temp,
+	lcp: temp,
+	fcp: temp,
+	ttfb: temp,
+};
+
+export const coreVitals = (): void => {
+	type CoreVitalsArgs = {
+		name: string;
+		value: number;
+	};
+
+	const getValue = ({ name, value }: CoreVitalsArgs): void => {
+		switch (name) {
+			case 'FCP':
+				jsonData.fcp = parseFloat(value.toFixed(6));
+				break;
+			case 'CLS':
+				jsonData.cls = parseFloat(value.toFixed(6));
+				break;
+			case 'LCP':
+				jsonData.lcp = parseFloat(value.toFixed(6));
+				break;
+			case 'FID':
+				jsonData.fid = parseFloat(value.toFixed(6));
+				break;
+			case 'TTFB':
+				jsonData.ttfb = parseFloat(value.toFixed(6));
+				break;
+		}
+
+		// If CLS has been calculated
+		if (jsonData.cls !== null) {
+			fetch(
+				'https://performance-events.guardianapis.com/core-web-vitals',
```

Comment:
I suppose we ought to configure this value so it works seamlessly between PROD and CODE/DEV. Not immediately sure where because typically this kind of thing comes from the CAPI data from frontend, but that doesn't seem wise for this. Maybe there's already somewhere sensible to do this in DCR directly. Doesn't seem worth putting in AWS parameter store, just for this.